code/modules/mining/machine_redemption.dm

```diff
@@ -236,8 +246,17 @@
 	else if (materials.on_hold())
 		data["disconnected"] = "mineral withdrawal is on hold"
 
+	var/obj/item/card/id/card
+	if(isliving(user))
+		var/mob/living/customer = user
+		card = customer.get_idcard(TRUE)
+	data["user"] = list()
+	if(card?.registered_account)
+		data["user"]["name"] = card.registered_account.account_holder
+		data["user"]["cash"] = card.registered_account.account_balance
```

Comment:
All of this should be inside the `isliving` check, then you define `card` inside there